mdm/checkin.go

```diff
@@ -27,6 +27,13 @@ func (svc *MDMService) Checkin(ctx context.Context, event CheckinEvent) error {
 		return errors.Wrap(err, "get checkin topic from message")
 	}
 
+	if topic == AuthenticateTopic {
```

Comment:
Probably doesn't make much of a difference, but `TokenUpdate` is when you really know if an enrollment happened.

Reply:
This is true. However since this service doesn't know which `TokenUpdate` has happened (the first, or any subsequent) it could potentially clear the queue in the midst of normal MDM operation if we checked on token update. `Authenticate` only happens during initial enrollment.

Separately I considered clearing on a `Checkout` event but decided against it for forensics considerations. Also separately I considered appending the queued commands to the "Failed" queue for forensics but decided that'd be far-fetched.